discojs/discojs-core/src/models/gpt/evaluate.ts

```diff
@@ -1,50 +1,58 @@
 import * as tf from '@tensorflow/tfjs'
 
+interface DataPoint extends tf.TensorContainerObject {
+  xs: tf.Tensor2D,
+  ys: tf.Tensor3D,
+}
+
 export default async function evaluate (
   model: tf.LayersModel,
-  dataset: tf.data.Dataset<{ xs: tf.Tensor, ys: tf.Tensor }>
+  dataset: tf.data.Dataset<{ xs: tf.Tensor, ys: tf.Tensor }>,
```

Comment:
no need to cast anymore in here, making the function safer

```suggestion
  dataset: tf.data.Dataset<DataPoint>,
```